src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederatedWorkerHandler.java

```diff
@@ -280,8 +281,9 @@ private FederatedResponse execInstruction(FederatedRequest request) {
 		ExecutionContext ec = _ecm.get(request.getTID());
 		BasicProgramBlock pb = new BasicProgramBlock(null);
 		pb.getInstructions().clear();
-		pb.getInstructions().add(InstructionParser
-			.parseSingleInstruction((String)request.getParam(0)));
+		Instruction receivedInstruction = InstructionParser
+				.parseSingleInstruction((String)request.getParam(0));
+		pb.getInstructions().add(receivedInstruction);
```

Comment:
I don't see what this change does.

Reply:
It does not do anything differently. I thought at some point that I would need to handle the receivedInstruction, but then I decided against it. I could change it back like it was before and the code would work the same way, so right now it is just a question of readability: is the new version easier to read for you, @Baunsgaard ?